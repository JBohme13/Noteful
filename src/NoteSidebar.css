.folder {
    height: 80px;
    text-align: center;
    margin-top: 2.5px;
    margin-bottom: 2.5px;
    border: 1px black solid;
    display: flex;
    flex-direction: column;
}

.folder-link {
    margin-top: auto;
    margin-bottom: auto;
}

#selected {
    background-color: rgb(52, 78, 221);
}

.back-button {
    margin: 5px;
    width: 80%;
    height: 32px;
}