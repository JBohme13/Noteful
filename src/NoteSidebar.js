import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import NotefulContext from './NotefulContext'
import FolderError from './FolderError'
import './NoteSidebar.css'

export default class NoteSidebar extends Component {
    static defaultProps = {
        match: {
          params: {}
        }
      }
    static contextType = NotefulContext;
    render() {
        const value= this.context;
        const folders = value.folders;
        const folderId = value.folderId;
        const selectedFolder = folders.map((folder, i) =>
        folder.id === folderId ? 
            <FolderError>
              <div key={i} className='folder' id='selected'>
                <Link 
                  className='folder-link' 
                  to={`/folders/${folder.id}`}
                  onClick={() => value.setFolderId(folder.id)}
                >
                  {folder.name}
                </Link>
              </div>
            </FolderError> : '')
        return(
            <div className='sidebar-container'>
              <div>
                { selectedFolder }
              </div>
              <button 
                id='back-button'
                onClick={() => value.history.push('/')}
              >
                  Go Back
              </button>
            </div>
        )
    }
}