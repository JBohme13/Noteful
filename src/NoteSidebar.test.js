import React from 'react';
import ReactDOM from 'react-dom';
import NoteSidebar from './NoteSidebar';
import renderer from 'react-test-renderer'

describe('NoteSidebar component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<NoteSidebar />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<NoteSidebar />)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
});