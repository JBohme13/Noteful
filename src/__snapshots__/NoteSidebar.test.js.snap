// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteSidebar component renders the UI as expected 1`] = `
<div
  className="sidebar-container"
>
  <div />
  <button
    id="back-button"
    onClick={[Function]}
  >
    Go Back
  </button>
</div>
`;
